File: src/Pascal's Triangle II/Pascal's Triangle II.py
class Solution:
	# @return a list of integers
	def getRow(self, rowIndex):
	ret, pre = [ 1 for i in range(rowIndex+1) ], 1                         
	if rowIndex < 2 : return ret 
	
	for row in range(3, rowIndex+2):
		for i in range(1, row):
			temp = ret[i]
			if i == row-1 : ret[i] = 1 
			else: ret[i] = pre + ret[i]
			pre = temp
	return ret 

File: src/Pascal's Triangle II/Pascal's Triangle II.cpp
class Solution {
	public:
		vector<int> getRow(int rowIndex) {
			vector<int> ret(rowIndex+1, 1);
			if(rowIndex+1 <= 2) return ret;

			int row = 3, pre = 1;
			while(row <= rowIndex+1)
			{
				for (int i = 1; i < row; i++) 
				{
					int temp = ret[i];
					if(i == row-1) ret[i] = 1;
					else ret[i] = pre + ret[i];
					pre = temp;
				}
				row++;
			}
			return ret;
		}
};
